marquez/client.py

```diff
@@ -0,0 +1,236 @@
+            "owner": namespace_owner,
+        }
+        if namespace_description:
+            payload['description'] = namespace_description
+        return self.put_request(path, path_args, payload, **kwargs)
+
+    #  Datasources API
+    @handle_response
+    def list_datasources(self, limit=None, offset=None, **kwargs):
+        path = "datasources"
+        path_args = []
+        request_args = {}
+        if limit:
+            request_args['limit'] = limit
+        if offset:
+            request_args['offset'] = offset
+        return self.get_request(path, path_args, request_args, **kwargs)
+
+    @handle_response
+    def get_datasource(self, urn, **kwargs):
+        path = "datasources/{0}"
+        path_args = [urn]
+        return self.get_request(path, path_args, **kwargs)
+
+    @handle_response
+    def create_datasource(self, name, connection_url, **kwargs):
+        path = "datasources"
+        path_args = []
+        payload = {
+            "name": name,
+            "connectionUrl": connection_url
+        }
+        return self.post_request(path, path_args, payload, **kwargs)
+
+    #  Jobs API
+    @handle_response
+    def create_job(self, job_name, location, input_dataset_urns,
+                   output_dataset_urns, description=None, **kwargs):
+        path = "namespaces/{0}/jobs/{1}"
+        path_args = [self.namespace, job_name]
+        payload = {
+            "inputDatasetUrns": input_dataset_urns,
+            "outputDatasetUrns": output_dataset_urns,
+            "location": location
+        }
+
+        if description:
+            payload['description'] = description
+        return self.put_request(path, path_args, payload, **kwargs)
+
+    @handle_response
+    def get_job(self, job_name, **kwargs):
+        path = "namespaces/{0}/jobs/{1}"
+        path_args = [self.namespace, job_name]
+        if 'timeout' not in kwargs:
+            kwargs['timeout'] = self.timeout
+        return self.get_request(path, path_args, **kwargs)
+
+    @handle_response
+    def get_jobs(self, **kwargs):
+        path = "namespaces/{0}/jobs/"
+        path_args = [self.namespace]
+        return self.get_request(path, path_args, **kwargs)
+
+    @handle_response
+    def create_job_run(self, job_name, job_run_args,
+                       nominal_start_time=None,
+                       nominal_end_time=None, **kwargs):
+        path = "namespaces/{0}/jobs/{1}/runs"
+        path_args = [self.namespace, job_name]
+        payload = {
+            "nominalStartTime": nominal_start_time,
+            "nominalEndTime": nominal_end_time,
+            "runArgs": job_run_args
+        }
+        return self.post_request(path, path_args, payload, **kwargs)
+
+    @handle_response
+    def get_job_run(self, job_run_id, **kwargs):
+        path = "jobs/runs/{0}"
+        path_args = [job_run_id]
+        return self.get_request(path, path_args, **kwargs)
+
+    def mark_job_run_running(self, job_run_id, **kwargs):
```

Comment:
_Minor_: `mark_job_run_running` **->** `mark_job_run_as_running`

Reply:
This would introduce an API change (here and elsewhere), albeit a minor.
Heads-up to @roaraya8 .

Reply:
Fixed.